src/main/java/core/basesyntax/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        validator.validateOperation(fruitTransaction);
+        String fruit = fruitTransaction.getFruit();
+        validator.validateFruit(fruit);
+        int suppliedAmount = fruitTransaction.getQuantity();
+        validator.validateAmount(suppliedAmount, FruitTransaction.Operation.SUPPLY);
+
+        Optional<Integer> currentBalance = Optional.ofNullable(Storage.fruitStorage
```

Comment:
Using `Optional` for checking if a value is present in the map is not necessary and can be simplified. You can directly use `Storage.fruitStorage.getOrDefault(fruit, 0)` to get the current value or default to 0 if the fruit is not in the storage.